### PR TITLE
CP-50259 simplify parsing size with kib, mib, etc suffix

### PR DESCRIPTION
**Pull request by lindig**
Simplify the implementation.

### CODE REVIEW

**Review by psafont — APPROVED** [review]

File: ocaml/xapi-cli-server/record_util.ml [hunk 1]
```diff
@@ -1075,89 +1075,35 @@ let domain_type_of_string x =
...
-         suffix)"
-        field
+  try bytes_of_string x
+  with Invalid_argument _ ->
+    record_failure
+      "Failed to parse field '%s': expecting an integer (possibly with suffix \
+       KiB, MiB, GiB, TiB), got '%s'"
+      field x
 
 (* Vincent's random mac utils *)
```
Comment (last-genius) [inline ocaml/xapi-cli-server/record_util.ml:1110]: I'm not sure this Vincent is here anymore, do we want to keep this?

Comment (lindig) [reply ocaml/xapi-cli-server/record_util.ml:1110]: Don't have to. I was not not sure what this is about.

Comment (psafont) [reply ocaml/xapi-cli-server/record_util.ml:1110]: He's sitting right next to me, though 😅

File: ocaml/xapi-cli-server/record_util.ml [hunk 2]
```diff
@@ -1075,89 +1075,35 @@ let domain_type_of_string x =
...
-
-      let alldigit = ref true and i = ref (String.length s - 1) in
-      while !alldigit && !i > 0 do
-        alldigit := Astring.Char.Ascii.is_digit s.[!i] ;
-        decr i
-      done ;
-      if !alldigit then
-        record_failure
-          "Failed to parse field '%s': number too big (maximum = %Ld TiB)" field
-          max_size_TiB
```
Comment (last-genius) [inline ocaml/xapi-cli-server/record_util.ml:1102]: Is it not important anymore to distinguish between "not an integer - couldn't parse" and "integer - but too big"? Additionally, we could overflow above when multiplying ** 1024 several times, do we want to catch and report that?

File: ocaml/xapi-cli-server/record_util.ml [hunk 3]
```diff
@@ -1075,89 +1075,35 @@ let domain_type_of_string x =
...
-            1024L ** 1024L ** 1024L
-        | "TiB" ->
-            1024L ** 1024L ** 1024L ** 1024L
-        | x ->
-            record_failure
-              "Failed to parse field '%s': Unknown suffix: '%s' (try KiB, MiB, \
-               GiB or TiB)"
-              field x
-      in
-      (* FIXME: detect overflow *)
```
Comment (last-genius) [inline ocaml/xapi-cli-server/record_util.ml:1154]: see comment above re: overflow

**Review by last-genius — APPROVED** [review]

File: ocaml/xapi-cli-server/record_util.ml (hunk 2 above)
Comment (lindig) [reply ocaml/xapi-cli-server/record_util.ml:1102]: Overflow is detected and results in an exception, which is caught. That number is incredibly large, though.

File: ocaml/xapi-cli-server/record_util.ml (hunk 1 above)
Comment (robhoes) [reply ocaml/xapi-cli-server/record_util.ml]: Definitely a different Vincent ;)

**Comment by robhoes** [comment]
Didn't the old function accept a space between the number and suffix, while the new one does not?

**Comment by lindig** [comment]
Reading the code, I believe you are right that the splitting allows a space. It's a bit awkward though, because this gets in the way of shell splitting words - so you would have quote that in a shell. Let me see if this can be accommodated.

**Pull request merged**